README.md
# BReco
In the first version, BReco package contains seven function that help you to download environmental datasets for Brazil.
You can to download data according with the geographycal level of interest and choose if the data must be presented in a data frame with panel data format or cross-sectional format.
You can to download data about:
* Fire focus (using the *get_fires* function);
* Biomass loss, tree cover loss and greenhouse gas emissions (using the function *get_gfw*);
* Annual average potential evapotranspiration (using the function *get_pet*);
* Annual average temperature (using the function *get_temperature*);
* Annual average rainfall (using the function *get_rainfall*);
* Anual greenhouse gas emissions (using the function *get_seeg*);
* Increase in deforestation, non-forest area, forest area, hydrography, cloud cover and deforested area in the Brazilian Legal Amazon (using the function *get_prodes*).

You can to acess the manual in https://trello-attachments.s3.amazonaws.com/5ea83f462064047eed09f846/5ea83fc7d5707865983b4f14/2e29079f591b9e00421f0db45206f71d/BReco_0.1.0.pdf


NEWS.md
## version 0.1.0

---

This is the first version of this package. In this version, the BReco package allows you to:

* Download data on greenhouse gas emissions from seeg.eco.br and clean this data.
* Download data on annual fire focus in brazilian municipalities.
* Download data on deforestation in the municipalities of the brazillian Legal Amazon.
* Download data on annual average precipitation, temperature and potential evapotranspiration in brazilian municipalities.
* Download data on tree cover loss, biomass loss and CO2 emissions in brazilian municipalities.
* The package also allows you to download the datasets in a panel data format or in a cross-sectional format.


R/get_pet.R
get_pet <- function(level = 'municipality', panel = FALSE){
  # The code below help to avoid the note "no visible binding for global variable [variable name]"
  CD_GEOCMU <- cod.state <- sigla.state <- country <- v1 <- evapotranspiration <- NULL
  if(level == 'municipality'){
    message('Please, wait for the data to download!')
    dt = openxlsx::read.xlsx('https://trello-attachments.s3.amazonaws.com/5ea83f462064047eed09f846/5ea83fc7d5707865983b4f14/12c62131af3179cbb4f21903ecd652b1/PET_1901_2012.xlsx')
  }
  if(level == 'state'){
    message('Please, wait for the data to download!')
    dt = openxlsx::read.xlsx('https://trello-attachments.s3.amazonaws.com/5ea83f462064047eed09f846/5ea83fc7d5707865983b4f14/12c62131af3179cbb4f21903ecd652b1/PET_1901_2012.xlsx')
    dt <- dplyr::mutate(dt, cod.state = substr(CD_GEOCMU, 1, 2),
                        sigla.state = ifelse(cod.state == '11', 'RO' , 'DF'),
                        sigla.state = ifelse(cod.state == '12', 'AC' , sigla.state), sigla.state = ifelse(cod.state == '13', 'AM' , sigla.state),
                        sigla.state = ifelse(cod.state == '14', 'RR' , sigla.state), sigla.state = ifelse(cod.state == '15', 'PA' , sigla.state),
                        sigla.state = ifelse(cod.state == '16', 'AP' , sigla.state), sigla.state = ifelse(cod.state == '17', 'TO' , sigla.state),
                        sigla.state = ifelse(cod.state == '21', 'MA' , sigla.state), sigla.state = ifelse(cod.state == '22', 'PI' , sigla.state),
                        sigla.state = ifelse(cod.state == '23', 'CE' , sigla.state), sigla.state = ifelse(cod.state == '24', 'RN' , sigla.state),
                        sigla.state = ifelse(cod.state == '25', 'PB' , sigla.state), sigla.state = ifelse(cod.state == '26', 'PE' , sigla.state),
                        sigla.state = ifelse(cod.state == '27', 'AL' , sigla.state), sigla.state = ifelse(cod.state == '28', 'SE' , sigla.state),
                        sigla.state = ifelse(cod.state == '29', 'BA' , sigla.state), sigla.state = ifelse(cod.state == '31', 'MG' , sigla.state),
                        sigla.state = ifelse(cod.state == '32', 'ES' , sigla.state), sigla.state = ifelse(cod.state == '33', 'RJ' , sigla.state),
                        sigla.state = ifelse(cod.state == '35', 'SP' , sigla.state), sigla.state = ifelse(cod.state == '41', 'PR' , sigla.state),
                        sigla.state = ifelse(cod.state == '42', 'SC' , sigla.state), sigla.state = ifelse(cod.state == '43', 'RS' , sigla.state),
                        sigla.state = ifelse(cod.state == '50', 'MS' , sigla.state), sigla.state = ifelse(cod.state == '51', 'MT' , sigla.state),
                        sigla.state = ifelse(cod.state == '52', 'GO' , sigla.state))
    dt <- dplyr::group_by(dt, sigla.state, cod.state)
    dt <- dplyr::summarise_at(dt, 4:115, mean)
  }
  if(level == 'country'){
    message('Please, wait for the data to download!')
    dt = openxlsx::read.xlsx('https://trello-attachments.s3.amazonaws.com/5ea83f462064047eed09f846/5ea83fc7d5707865983b4f14/12c62131af3179cbb4f21903ecd652b1/PET_1901_2012.xlsx')
    dt <- dplyr::mutate(dt, country = 'Brazil')
    dt <- dplyr::group_by(dt, country)
    dt <- dplyr::summarise_at(dt, 4:115, mean)
  }
  if(panel == TRUE){
    dt<- tidyr::gather(dt, v1, evapotranspiration, 'PET_1901':'PET_2012')
    dt <- dplyr::mutate(dt, year = substr(v1, 5,8))
    dt <- dplyr::select(dt, -v1)
  }
  if(level %in% c('municipality', 'state', 'country') == FALSE | panel %in% c(T, F) == FALSE){
    stop('\n
    -------------------------------------------------------- \n
    Please, enter a valid value to level or panel parameter! \n
    -------------------------------------------------------- \n')
  }
  warning('\n
  -------------------------------------------------- \n
  Please, cite https://www.globalclimatemonitor.org/ \n
  and \n
  CAMARILLO-NARANJO, Juan Mariano et al. The global \n
  climate monitor system: from climate data-handling \n
  to knowledge dissemination. International journal \n
  of digital earth, v. 12, n. 4, p. 394-414, 2019.
  -------------------------------------------------- ')
  return(dt)
}




R/get_seeg.R
get_seeg <- function(state = NULL, activity = NULL, gas = NULL, type_data = NULL, panel = FALSE){
  # The code below help to avoid the note "no visible binding for global variable [variable name]"
  Estado <- setor <- Gas <- Emissao_Remocao <- year <- value <- NULL
  if(is.null(activity) == FALSE){
    if(activity %in% c('agriculture', 'energy', 'land use change', 'industry', 'waste') == FALSE){
      stop('\n
        ------------------------------- \n
        Please, enter a valid activity! \n
        -------------------------------')
    }
  }
  if(is.null(state) == FALSE){
    if(state %in% c('AC', 'AM', 'PA', 'RO', 'RR', 'AP', 'TO', 'MA', 'PI', 'CE', 'RN', 'PB', 'PE', 'AL', 'SE', 'BA', 'ES', 'MG', 'SP', 'RJ', 'PR', 'SC', 'RS', 'MS', 'MT', 'GO') == FALSE){
      stop('\n
      ------------------------------------------- \n
      Please, enter a valid brazilian state code! \n
      -------------------------------------------')
    }
  }
  if(is.null(gas) == FALSE){
    if (gas %in% c("CO2e (t) GTP-AR2", "CO2e (t) GTP-AR4", "CO2e (t) GTP-AR5", "CO2e (t) GWP-AR2",
                   "CO2e (t) GWP-AR4", "CO2e (t) GWP-AR5", "CO2 (t)", "CO (t)", "CH4 (t)", "NOx (t)", "N2O (t)", "COVNM (t)", "CF4 (t)", "C2F6 (t)",
                   "SF6 (t)", "HFC-23 (t)", "HFC-32 (t)", "HFC-134a (t)", "HFC-125 (t)", "HFC-143a (t)", "HFC-152a (t)", "NOX (t)" ) == FALSE){
      stop('\n
        -------------------------- \n
        Pleace, enter a valid gas! \n
        --------------------------')
    }
  }
  if(is.null(type_data) == FALSE){
    if(type_data %in% c('emissions', 'removal') == FALSE){
      stop('\n
      ------------------------------------------------------ \n
      Pleace, choose a opition between emissions or removal! \n
      ------------------------------------------------------')
    }
  }
  print('----- Wait for the data to download! -----')
  data <- openxlsx::read.xlsx('https://github.com/Helson-Gomes/dadostse/raw/master/SEEG.xlsx')
  data <- dplyr::rename(data, setor = 1)
  if(is.null(state) == FALSE){
    data<- subset(data, Estado == state)
  }
  if(is.null(activity) == FALSE){
    if(activity == 'agriculture'){
      data <- subset(data, setor == 'Agropecuaria')
    }
    if(activity == 'energy'){
      data <- subset(data, setor == 'Energia')
    }
    if(activity == 'land use change'){
      data <- subset(data, setor == 'Mudanca de Uso da Terra')
    }
    if(activity == 'industry'){
      data <- subset(data, setor == 'Processos Industriais')
    }
    if(activity == 'waste'){
      data <- subset(data, setor == 'Residuos')
    }
  }
  if(is.null(gas) == FALSE){
    data <- subset(data, Gas == gas)
  }
  if(is.null(type_data) == FALSE){
    if (type_data == 'emissions'){
      data <- subset(data, Emissao_Remocao == "Emissao")
    }
    if (type_data == 'removal'){
      data <- subset(data, Emissao_Remocao == "Remocao")
    }
  }
  if(panel == TRUE){
    data <- tidyr::gather(data, year, value, '1970':'2018')
    data <- dplyr::select(data, 1,7,8,9,12,13)
  }
  if(panel == FALSE){
    data <- dplyr::select(data, 1, 7, 8, 9, 12:60)
  }
  if(panel %in% c(FALSE, TRUE) == FALSE){
    stop('\n
      --------------------------------------------- \n
      Please, choose panel = FALSE or panel = TRUE! \n
      ---------------------------------------------')
  }
  warning('\n
  ------------------------ \n
  Please, cite seeg.eco.br \n
  ------------------------')
  return(data)
}



R/get_fires.R
get_fires <- function(level = NULL, panel = FALSE){
  # The code below help to avoid the note "no visible binding for global variable [variable name]"
  CD_GEOCMU <-  cod.state <- sigla.state <- year <- country <- fires <- v1 <- NULL


  temp <- tempfile()
  if(is.null(level) == TRUE){
    message('Pleace, wait for the data to download!')
    dt <- openxlsx::read.xlsx('https://trello-attachments.s3.amazonaws.com/5ea83f462064047eed09f846/5ea83fc7d5707865983b4f14/16378a8eeeb8d1651493fe4e9606cf51/INCENDIOS_2001_2019.xlsx')
  }else{
    if(level == 'municipality'){
      message('Pleace, wait for the data to download!')
      dt <- openxlsx::read.xlsx('https://trello-attachments.s3.amazonaws.com/5ea83f462064047eed09f846/5ea83fc7d5707865983b4f14/16378a8eeeb8d1651493fe4e9606cf51/INCENDIOS_2001_2019.xlsx')
    }
    if(level == 'state'){
      message('Pleace, wait for the data to download!')
      dt <- openxlsx::read.xlsx('https://trello-attachments.s3.amazonaws.com/5ea83f462064047eed09f846/5ea83fc7d5707865983b4f14/16378a8eeeb8d1651493fe4e9606cf51/INCENDIOS_2001_2019.xlsx')
      dt <- dplyr::mutate(dt, cod.state = substr(CD_GEOCMU, 1, 2),
                          sigla.state = ifelse(cod.state == '11', 'RO' , 'DF'),
                          sigla.state = ifelse(cod.state == '12', 'AC' , sigla.state), sigla.state = ifelse(cod.state == '13', 'AM' , sigla.state),
                          sigla.state = ifelse(cod.state == '14', 'RR' , sigla.state), sigla.state = ifelse(cod.state == '15', 'PA' , sigla.state),
                          sigla.state = ifelse(cod.state == '16', 'AP' , sigla.state), sigla.state = ifelse(cod.state == '17', 'TO' , sigla.state),
                          sigla.state = ifelse(cod.state == '21', 'MA' , sigla.state), sigla.state = ifelse(cod.state == '22', 'PI' , sigla.state),
                          sigla.state = ifelse(cod.state == '23', 'CE' , sigla.state), sigla.state = ifelse(cod.state == '24', 'RN' , sigla.state),
                          sigla.state = ifelse(cod.state == '25', 'PB' , sigla.state), sigla.state = ifelse(cod.state == '26', 'PE' , sigla.state),
                          sigla.state = ifelse(cod.state == '27', 'AL' , sigla.state), sigla.state = ifelse(cod.state == '28', 'SE' , sigla.state),
                          sigla.state = ifelse(cod.state == '29', 'BA' , sigla.state), sigla.state = ifelse(cod.state == '31', 'MG' , sigla.state),
                          sigla.state = ifelse(cod.state == '32', 'ES' , sigla.state), sigla.state = ifelse(cod.state == '33', 'RJ' , sigla.state),
                          sigla.state = ifelse(cod.state == '35', 'SP' , sigla.state), sigla.state = ifelse(cod.state == '41', 'PR' , sigla.state),
                          sigla.state = ifelse(cod.state == '42', 'SC' , sigla.state), sigla.state = ifelse(cod.state == '43', 'RS' , sigla.state),
                          sigla.state = ifelse(cod.state == '50', 'MS' , sigla.state), sigla.state = ifelse(cod.state == '51', 'MT' , sigla.state),
                          sigla.state = ifelse(cod.state == '52', 'GO' , sigla.state))
      dt <- dplyr::group_by(dt, sigla.state)
      dt<- dplyr::summarise_at(dt, 2:20, sum)
    }
    if(level == 'country'){
      message('Pleace, wait for the data to download!')
      dt <- openxlsx::read.xlsx('https://trello-attachments.s3.amazonaws.com/5ea83f462064047eed09f846/5ea83fc7d5707865983b4f14/16378a8eeeb8d1651493fe4e9606cf51/INCENDIOS_2001_2019.xlsx')
      dt<- dplyr::mutate(dt, country = 'Brazil')
      dt<-dplyr::group_by(dt, country)
      dt <- dplyr::summarise_at(dt, 2:20, sum)
    }
  }
  if(panel == TRUE){
    dt<-tidyr::gather(dt, v1, fires, 'FIRE_2001':'FIRE_2019')
    dt <- dplyr::mutate(dt, year = substr(v1, 6, 9))
    dt <- dplyr::select(dt, -v1)
  }
  warning('\n
  -------------------------------------------------------------------------------------- \n
  Please, cite https://earthdata.nasa.gov/earth-observation-data/near-real-time/firms \n
  -------------------------------------------------------------------------------------- ')
  return(dt)
}


R/get_prodes.R
get_prodes <- function(year = NULL, state = NULL){
  # The code below help to avoid the note "no visible binding for global variable [variable name]"
  Estado <- NULL
  if(year %in% c(2002:2018) == FALSE){
    stop('\n
    -------------------------------------------- \n
    Please, choose a year between 2002 and 2018! \n
    --------------------------------------------')
  } else{
    dt <- read.csv2(paste(sprintf('http://www.dpi.inpe.br/prodesdigital/tabelatxt.php?ano=%i&estado=&ordem=DESMATAMENTO%i&type=tabela&output=txt&', year, year)), sep = ',', dec = '.', encoding = 'latin1')
    if(is.null(state) == FALSE){
      if(state %in% c('AC', 'AM', 'AP', 'MA', 'MT', 'PA', 'RO', 'RR', 'TO') == FALSE){
        stop('\n
        ----------------------------------------------- \n
        Please, choose a valid Legal Amazon state code! \n
        -----------------------------------------------')
      } else{
        dt <- dplyr::filter(dt, Estado == state)
      }
    }
  }
  warning('\n
  --------------------------------------------------------------------- \n
  Please, cite http://www.dpi.inpe.br/prodesdigital/prodesmunicipal.php \n
  ---------------------------------------------------------------------')
  return(dt)
}




R/get_rainfall.R
get_rainfall <- function(level = 'municipality', panel = FALSE){
  # The code below help to avoid the note "no visible binding for global variable [variable name]"
  CD_GEOCMU <- cod.state <- sigla.state <- country <- v1 <- rainfall <- NULL
  temp <- tempfile()
  if(level == "municipality"){
    message("Please, wait for the data to download!")
    dt <- openxlsx::read.xlsx("https://trello-attachments.s3.amazonaws.com/5ea83f462064047eed09f846/5ea83fc7d5707865983b4f14/2fe3829d60a70197d0b44337440348a2/PEECIPITACAO_1901_2019.xlsx")
  }
  if(level == 'state'){
    message('Please, wait for the data to download!')
    dt <- openxlsx::read.xlsx("https://trello-attachments.s3.amazonaws.com/5ea83f462064047eed09f846/5ea83fc7d5707865983b4f14/2fe3829d60a70197d0b44337440348a2/PEECIPITACAO_1901_2019.xlsx")
    dt <- dplyr::mutate(dt, cod.state = substr(CD_GEOCMU, 1, 2),
                        sigla.state = ifelse(cod.state == '11', 'RO' , 'DF'),
                        sigla.state = ifelse(cod.state == '12', 'AC' , sigla.state), sigla.state = ifelse(cod.state == '13', 'AM' , sigla.state),
                        sigla.state = ifelse(cod.state == '14', 'RR' , sigla.state), sigla.state = ifelse(cod.state == '15', 'PA' , sigla.state),
                        sigla.state = ifelse(cod.state == '16', 'AP' , sigla.state), sigla.state = ifelse(cod.state == '17', 'TO' , sigla.state),
                        sigla.state = ifelse(cod.state == '21', 'MA' , sigla.state), sigla.state = ifelse(cod.state == '22', 'PI' , sigla.state),
                        sigla.state = ifelse(cod.state == '23', 'CE' , sigla.state), sigla.state = ifelse(cod.state == '24', 'RN' , sigla.state),
                        sigla.state = ifelse(cod.state == '25', 'PB' , sigla.state), sigla.state = ifelse(cod.state == '26', 'PE' , sigla.state),
                        sigla.state = ifelse(cod.state == '27', 'AL' , sigla.state), sigla.state = ifelse(cod.state == '28', 'SE' , sigla.state),
                        sigla.state = ifelse(cod.state == '29', 'BA' , sigla.state), sigla.state = ifelse(cod.state == '31', 'MG' , sigla.state),
                        sigla.state = ifelse(cod.state == '32', 'ES' , sigla.state), sigla.state = ifelse(cod.state == '33', 'RJ' , sigla.state),
                        sigla.state = ifelse(cod.state == '35', 'SP' , sigla.state), sigla.state = ifelse(cod.state == '41', 'PR' , sigla.state),
                        sigla.state = ifelse(cod.state == '42', 'SC' , sigla.state), sigla.state = ifelse(cod.state == '43', 'RS' , sigla.state),
                        sigla.state = ifelse(cod.state == '50', 'MS' , sigla.state), sigla.state = ifelse(cod.state == '51', 'MT' , sigla.state),
                        sigla.state = ifelse(cod.state == '52', 'GO' , sigla.state))
    dt <- dplyr::group_by(dt, sigla.state, cod.state)
    dt <- dplyr::summarise_at(dt, 5:123, mean)
  }
  if(level == 'country'){
    message('Please, wait for the data to download!')
    dt <- openxlsx::read.xlsx("https://trello-attachments.s3.amazonaws.com/5ea83f462064047eed09f846/5ea83fc7d5707865983b4f14/2fe3829d60a70197d0b44337440348a2/PEECIPITACAO_1901_2019.xlsx")
    dt <- dplyr::mutate(dt, country = 'Brazil')
    dt <- dplyr::group_by(dt, country)
    dt <- dplyr::summarise_at(dt, 5:123, mean)
  }
  if(panel == TRUE){
    dt <- tidyr::gather(dt, v1, rainfall, 'PREC_1901':'PREC_2019')
    dt <- dplyr::mutate(dt, year = substr(v1, 6,9))
    dt <- dplyr::select(dt, -v1)
  }
  if(level %in% c('municipality', 'state', 'country') == FALSE | panel %in% c(T, F) == FALSE){
    stop('\n
    -------------------------------------------------------- \n
    Please, enter a valid value to level or panel parameter! \n
    -------------------------------------------------------- \n')
  }
  warning('\n
  -------------------------------------------------- \n
  Please, cite https://www.globalclimatemonitor.org/ \n
  and \n
  CAMARILLO-NARANJO, Juan Mariano et al. The global \n
  climate monitor system: from climate data-handling \n
  to knowledge dissemination. International journal \n
  of digital earth, v. 12, n. 4, p. 394-414, 2019.
  -------------------------------------------------- ')
  return(dt)
}




R/get_temperature.R
get_temperature <- function(level = 'municipality', panel = FALSE){
  # The code below help to avoid the note "no visible binding for global variable [variable name]"
  CD_GEOCMU <- cod.state <- sigla.state <- country <- v1 <- temperature <- NULL
  if(level == 'municipality'){
    message('Please, wait for the data to download!')
    dt <- openxlsx::read.xlsx('https://trello-attachments.s3.amazonaws.com/5ea83f462064047eed09f846/5ea83fc7d5707865983b4f14/770f4bdff850c0b9c35113e131999ff2/TEMPERATURA_1901_2019.xlsx')
  }
  if(level == 'state'){
    message('Please, wait for the data to download!')
    dt <- openxlsx::read.xlsx('https://trello-attachments.s3.amazonaws.com/5ea83f462064047eed09f846/5ea83fc7d5707865983b4f14/770f4bdff850c0b9c35113e131999ff2/TEMPERATURA_1901_2019.xlsx')
    dt <- dplyr::mutate(dt, cod.state = substr(CD_GEOCMU, 1, 2),
                        sigla.state = ifelse(cod.state == '11', 'RO' , 'DF'),
                        sigla.state = ifelse(cod.state == '12', 'AC' , sigla.state), sigla.state = ifelse(cod.state == '13', 'AM' , sigla.state),
                        sigla.state = ifelse(cod.state == '14', 'RR' , sigla.state), sigla.state = ifelse(cod.state == '15', 'PA' , sigla.state),
                        sigla.state = ifelse(cod.state == '16', 'AP' , sigla.state), sigla.state = ifelse(cod.state == '17', 'TO' , sigla.state),
                        sigla.state = ifelse(cod.state == '21', 'MA' , sigla.state), sigla.state = ifelse(cod.state == '22', 'PI' , sigla.state),
                        sigla.state = ifelse(cod.state == '23', 'CE' , sigla.state), sigla.state = ifelse(cod.state == '24', 'RN' , sigla.state),
                        sigla.state = ifelse(cod.state == '25', 'PB' , sigla.state), sigla.state = ifelse(cod.state == '26', 'PE' , sigla.state),
                        sigla.state = ifelse(cod.state == '27', 'AL' , sigla.state), sigla.state = ifelse(cod.state == '28', 'SE' , sigla.state),
                        sigla.state = ifelse(cod.state == '29', 'BA' , sigla.state), sigla.state = ifelse(cod.state == '31', 'MG' , sigla.state),
                        sigla.state = ifelse(cod.state == '32', 'ES' , sigla.state), sigla.state = ifelse(cod.state == '33', 'RJ' , sigla.state),
                        sigla.state = ifelse(cod.state == '35', 'SP' , sigla.state), sigla.state = ifelse(cod.state == '41', 'PR' , sigla.state),
                        sigla.state = ifelse(cod.state == '42', 'SC' , sigla.state), sigla.state = ifelse(cod.state == '43', 'RS' , sigla.state),
                        sigla.state = ifelse(cod.state == '50', 'MS' , sigla.state), sigla.state = ifelse(cod.state == '51', 'MT' , sigla.state),
                        sigla.state = ifelse(cod.state == '52', 'GO' , sigla.state))
    dt <- dplyr::group_by(dt, sigla.state)
    dt <- dplyr::summarise_at(dt, 5:123, mean)
  }
  if(level == 'country'){
    message('Please, wait for the data to download!')
    dt <- openxlsx::read.xlsx('https://trello-attachments.s3.amazonaws.com/5ea83f462064047eed09f846/5ea83fc7d5707865983b4f14/770f4bdff850c0b9c35113e131999ff2/TEMPERATURA_1901_2019.xlsx')
    dt <- dplyr::mutate(dt, country = 'Brazil')
    dt <- dplyr::group_by(dt, country)
    dt <- dplyr::summarise_at(dt, 5:123, mean)
  }
  if(panel == TRUE){
    dt <- tidyr::gather(dt, v1, temperature, 'TEMP_1901':'TEMP_2019')
    dt <- dplyr::mutate(dt, year = substr(v1, 6,9))
    dt <- dplyr::select(dt, -v1)
  }
  if(level %in% c('municipality', 'state', 'country') == FALSE | panel %in% c(T, F) == FALSE){
    stop('\n
    -------------------------------------------------------- \n
    Please, enter a valid value to level or panel parameter! \n
    -------------------------------------------------------- \n')
  }
  warning('\n
  -------------------------------------------------- \n
  Please, cite https://www.globalclimatemonitor.org/ \n
  and \n
  CAMARILLO-NARANJO, Juan Mariano et al. The global \n
  climate monitor system: from climate data-handling \n
  to knowledge dissemination. International journal \n
  of digital earth, v. 12, n. 4, p. 394-414, 2019.
  -------------------------------------------------- ')
  return(dt)
}




R/get_gfw.R
get_gfw <- function(type_data = NULL, level = NULL){
  if(is.null(type_data) == FALSE){
    if(is.null(level) == FALSE){
      if(level %in% c('Country', 'Subnational 1', 'Subnational 2') == FALSE){
        stop('
        ----------------------------------------- \n
        Please, choose a valid territorial level! \n
        -----------------------------------------')
      }
    } else{
      stop('
      ----------------------------------------- \n
      Please, choose a valid territorial level! \n
      -----------------------------------------')
    }
    if(type_data %in% c('co2 emissions', 'biomass loss', 'tree cover loss') == FALSE){
      stop('
      --------------------------------- \n
      Please, choose a valid data type! \n
      --------------------------------')
    }
    print('Please, wait for the data to download!')
    df <- openxlsx::read.xlsx('https://gfw2-data.s3.amazonaws.com/country-pages/country_stats/download/BRA.xlsx', sheet = paste(level, type_data))



  }else{
    stop('\n
    ---------------------------------- \n
    Please, choose a valide data type! \n
    ----------------------------------')
  }
  warning('\n
  ----------------------------------------------------------------------------------------------------\n
  Pleace, cite: \n
  Hansen, M. C., P. V. Potapov, R. Moore, M. Hancher, S. A. Turubanova, A. Tyukavina, D.\n
  Thau, S. V. Stehman, S. J. Goetz, T. R. Loveland, A. Kommareddy, A. Egorov, L. Chini, C. O. \n
  Justice, and J. R. G. Townshend. 2013. ???High-Resolution Global Maps of 21st-Century Forest \n
  Cover Change.??? Science 342 (15 November): 850???53. Data available on-line from: \n
  http://earthenginepartners.appspot.com/science-2013-global-forest. \n
  \n
  Zarin, D., Harris, N.L. et al. 2016. Can carbon emissions drop by 50% in five years? Global \n
  Change Biology, 22: 1336-1347. doi:10.1111/gcb.13153 \n
  and \n
  \n
  Global Administrative Areas Database, version 3.6. Available at http://gadm.org/ \n
  ----------------------------------------------------------------------------------------------------')
  return(df)
}


